### PR TITLE
update chess library

### PR DESCRIPTION
**Pull request by robertnurnberg**
This fixes two possible issues in pgn move parsing.

### CODE REVIEW

File: scoreWDLstat.cpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ class Analyze : public pgn::Visitor {
 
     void endPgn() override {
         board.set960(false);
-        board.setFen(STARTPOS);
+        board.setFen(chess::constants::STARTPOS);
```
Comment (Disservin) [inline scoreWDLstat.cpp:198]: no need for the `chess` prefix

**Comment by Disservin** [comment]
next would be also to make use of the `skipPgn(true)` functionality, allows for a bit easier pgn skipping

**Comment by robertnurnberg** [comment]
The pgn parsing from within the chess-library should be fixed now.

**Comment by Disservin** [comment]
@robertnurnberg I get the same scored position with these changes, you too?

As a side note it seems the parser is slightly slower at least for me? Not of much importance anyway.
```
master: Time taken: 8.539s
patch:  Time taken: 10.886s
```

**Comment by robertnurnberg** [comment]
Yes, I tested for non-functionality on a (small) test set of pgns.

**Pull request merged**